aws/cloudformation.yml

```diff
@@ -0,0 +1,699 @@
+                    proxy_set_header Upgrade $http_upgrade;
+                    proxy_set_header Connection $connection_upgrade;
+
+                    tcp_nodelay on;
+                  }
+
+                  error_page 500 501 502 503 504 /500.html;
+                }
+              mode: "000777"
+              owner: "ubuntu"
+              group: "ubuntu"
+
+            /var/tmp/docker-compose.yml:
+              content: !Sub |
+                  version: '3'
+                  services:
+                    web:
+                      build: .
+                      image: tootsuite/mastodon:${ReleaseVersion}
+                      restart: always
+                      env_file: .env.production
+                      command: bash -c "rm -f /mastodon/tmp/pids/server.pid; bundle exec rails s -p 3000 -b '0.0.0.0'"
+                      networks:
+                        - external_network
+                        - internal_network
+                      ports:
+                        - "127.0.0.1:3000:3000"
+                      volumes:
+                        - ./public/system:/mastodon/public/system
+
+                    streaming:
+                      build: .
+                      image: tootsuite/mastodon:${ReleaseVersion}
+                      restart: always
+                      env_file: .env.production
+                      command: yarn start
+                      networks:
+                        - external_network
+                        - internal_network
+                      ports:
+                        - "127.0.0.1:4000:4000"
+
+                    sidekiq:
+                      build: .
+                      image: tootsuite/mastodon:${ReleaseVersion}
+                      restart: always
+                      env_file: .env.production
+                      command: bundle exec sidekiq -q default -q mailers -q pull -q push
+                      networks:
+                        - external_network
+                        - internal_network
+                      volumes:
+                        - ./public/system:/mastodon/public/system
+                  networks:
+                    external_network:
+                    internal_network:
+                      internal: true
+              mode: "000777"
+              owner: "ubuntu"
+              group: "ubuntu"
+
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT2H
+
+  EC2InstanceProfile:
+    Type: 'AWS::IAM::InstanceProfile'
+    Properties:
+      Path: /
+      Roles:
+        - !Ref EC2Role
+
+  EC2Role:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: ec2.amazonaws.com
+            Action: 'sts:AssumeRole'
+      Path: /
+      Policies:
+        - PolicyName: Oracle-ec2
+          PolicyDocument:
+            Statement:
+              - Effect: Allow
+                Action: 's3:*'
```

Comment:
Same again for this wildcard

Also on L668, don't give us more than we need D:

Reply:
fixed.